ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/KeyValueParticipantStateWriter.scala

```diff
@@ -23,9 +23,14 @@ class KeyValueParticipantStateWriter(writer: LedgerWriter)(
   override def submitTransaction(
       submitterInfo: SubmitterInfo,
       transactionMeta: TransactionMeta,
-      transaction: SubmittedTransaction): CompletionStage[SubmissionResult] = {
+      transaction: SubmittedTransaction,
+  ): CompletionStage[SubmissionResult] = {
     val submission =
-      KeyValueSubmission.transactionToSubmission(submitterInfo, transactionMeta, transaction)
+      KeyValueSubmission.transactionToSubmission(
+        submitterInfo,
+        transactionMeta,
+        transaction.assertNoRelCid(cid => s"Unexpected relative contract id: $cid"),
```

Comment:
I would rather catch the relative contract ids in the transaction encoding. Pretty expensive traversing it just for this.

Reply:
The objective is to completly get ride of the relative contract id. I would prefer kvutils type does not reference at all relative contract ids and drop support of old contract ids scheme in the sandbox as soon as possible (that is 2/3 releases). 
So we could simply merge `SubmittedTransaction` and `ComittedTransaction` (using only absolute contract id) and get rid of the `assertNoRelCid`

If it is very urgent we can find a work arround to drop completly relative contract ids and still have the sandbox support old scheme (that is `#txId:nodeId`)

Reply:
sounds good to me. lets keep it and later clean up.